pkg/git/client.go

```diff
@@ -162,7 +163,7 @@ func (g *Client) writeCheck() (err error) {
 			config.RefSpec("+" + ref.Name() + ":" + ref.Name()),
 		},
 		Auth: g.auth,
-	}); pushErr != nil && pushErr == gogit.NoErrAlreadyUpToDate {
+	}); pushErr != nil && pushErr != gogit.NoErrAlreadyUpToDate {
 		return fmt.Errorf("write-check failed: %w", pushErr)
 	}
 
```

Comment:
```suggestion
	}); pushErr != nil && pushErr != gogit.NoErrAlreadyUpToDate {
		return fmt.Errorf("write-check failed: %w", pushErr)
	}
      // explicitly ignore other push errors
      pushErr = nil
```

Reply:
pushErr = nil shows the reader immediately that other pushErrs have no effect

Reply:
I see, all errors are returned except gogit.NoErrAlreadyUpToDate. So it's okay for me.